src/ol/disposable.js

```diff
@@ -0,0 +1,30 @@
+goog.provide('ol.Disposable');
+
+/**
+ * Objects that need to clean up after themselves.
+ * @constructor
+ */
+ol.Disposable = function() {};
+
+/**
+ * The object has already been disposed.
+ * @type {boolean}
+ * @private
+ */
+ol.Disposable.prototype.disposed_ = false;
+
+/**
+ * Clean up.
+ */
+ol.Disposable.prototype.dispose = function() {
```

Comment:
Not sure to understand whether calling `dispose()` is exclusively the responsability of the library or if the user should do it sometimes.
For example, the `ol.Map` class inherits from `ol.Disposable`. Is the user supposed to call the `dispose()` method when he wants to get ride of a map instance?